File: 프로그래머스/unrated/181935. 홀짝에 따라 다른 값 반환하기/홀짝에 따라 다른 값 반환하기.js
function solution(n) {
  let odd = Array.from({length:n},(_,i)=>i+1).filter(v=> v%2).reduce((a,c)=>a+c,0);
  let even = Array.from({length:n},(_,i)=>i+1).filter(v=> !(v%2)).reduce((a,c)=>a+ c**2,0)
  return n%2 ? odd : even
}

File: 프로그래머스/unrated/181935. 홀짝에 따라 다른 값 반환하기/홀짝에 따라 다른 값 반환하기.java
import java.util.stream.*;

class Solution {
    public int solution(int n) {
        if(n % 2 == 1){
            return IntStream.rangeClosed(1,n).filter(v->v%2 == 1).sum();
        }
        return IntStream.rangeClosed(1,n).filter(v->v%2 == 0).map(v-> v*v).sum();
    }
}